frontend/src/components/widgets/TextArea/TextArea.tsx

```diff
@@ -69,29 +69,46 @@ class TextArea extends React.PureComponent<Props, State> {
     })
   }
 
-  private onKeyPress = (e: React.KeyboardEvent<HTMLTextAreaElement>): void => {
-    if (e.key === "Enter" && e.ctrlKey && this.state.dirty) {
+  private handleSubmitEvent = (
+    e: React.KeyboardEvent<HTMLTextAreaElement>
+  ): void => {
+    const { key, metaKey, ctrlKey } = e
+    const { dirty } = this.state
+
+    if (key === "Enter" && (metaKey || ctrlKey) && dirty) {
       this.setWidgetValue({ fromUi: true })
     }
   }
 
+  private onKeyPress = (e: React.KeyboardEvent<HTMLTextAreaElement>): void => {
+    this.handleSubmitEvent(e)
+  }
+
+  private onKeyDown = (e: React.KeyboardEvent<HTMLTextAreaElement>): void => {
```

Comment:
Why do we need both keydown and keypress? Won't this submit twice?

Reply:
The metaKey is not being handled in the keyPress event

Reply:
I mean the metaKey it's only captured within keyDown event

Reply:
I see... but the question remains: this will cause the text area to submit twice, right? If so, we should fix that.

Reply:
We are preventing that using the 'dirty' check but i'll change it